src/app/lingua-puzzle/services/sentence/sentence.service.ts

```diff
@@ -0,0 +1,35 @@
+import { inject, Injectable } from '@angular/core';
+import { RoundService } from '../round/round.service';
+import { Sentence } from '../../../shared/types/http-data.interface';
+import { Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SentenceService {
+  private readonly roundService = inject(RoundService);
+  private sentences: Sentence[] = [];
+  public sentence = new Subject<Sentence>();
+  public sentenceIndex = 0;
+
+  constructor() {
+    this.roundService.round.subscribe((round) => {
+      this.sentences = round.words;
+      this.sentenceIndex = 0;
+      this.setSentence();
+    });
+  }
```

Comment:
make `sentences` a reactive depends on `roundService.round` then make `sentence` depend on a `sentences` and `roundIndex`